engine/src/main/java/org/terasology/rendering/cameras/ViewFrustum.java

```diff
@@ -147,35 +156,80 @@ public boolean intersects(AABB aabb) {
 
         for (int i = 0; i < 6; i++) {
             if (planes[i].getA() * (aabbVertices[0].x - cp.x) + planes[i].getB() * (aabbVertices[0].y - cp.y)
-                    + planes[i].getC() * (aabbVertices[0].z - cp.z) + planes[i].getD() > 0) {
+                + planes[i].getC() * (aabbVertices[0].z - cp.z) + planes[i].getD() > 0) {
                 continue;
             }
             if (planes[i].getA() * (aabbVertices[1].x - cp.x) + planes[i].getB() * (aabbVertices[1].y - cp.y)
-                    + planes[i].getC() * (aabbVertices[1].z - cp.z) + planes[i].getD() > 0) {
+                + planes[i].getC() * (aabbVertices[1].z - cp.z) + planes[i].getD() > 0) {
                 continue;
             }
             if (planes[i].getA() * (aabbVertices[2].x - cp.x) + planes[i].getB() * (aabbVertices[2].y - cp.y)
-                    + planes[i].getC() * (aabbVertices[2].z - cp.z) + planes[i].getD() > 0) {
+                + planes[i].getC() * (aabbVertices[2].z - cp.z) + planes[i].getD() > 0) {
                 continue;
             }
             if (planes[i].getA() * (aabbVertices[3].x - cp.x) + planes[i].getB() * (aabbVertices[3].y - cp.y)
-                    + planes[i].getC() * (aabbVertices[3].z - cp.z) + planes[i].getD() > 0) {
+                + planes[i].getC() * (aabbVertices[3].z - cp.z) + planes[i].getD() > 0) {
                 continue;
             }
             if (planes[i].getA() * (aabbVertices[4].x - cp.x) + planes[i].getB() * (aabbVertices[4].y - cp.y)
-                    + planes[i].getC() * (aabbVertices[4].z - cp.z) + planes[i].getD() > 0) {
+                + planes[i].getC() * (aabbVertices[4].z - cp.z) + planes[i].getD() > 0) {
                 continue;
             }
             if (planes[i].getA() * (aabbVertices[5].x - cp.x) + planes[i].getB() * (aabbVertices[5].y - cp.y)
-                    + planes[i].getC() * (aabbVertices[5].z - cp.z) + planes[i].getD() > 0) {
+                + planes[i].getC() * (aabbVertices[5].z - cp.z) + planes[i].getD() > 0) {
                 continue;
             }
             if (planes[i].getA() * (aabbVertices[6].x - cp.x) + planes[i].getB() * (aabbVertices[6].y - cp.y)
-                    + planes[i].getC() * (aabbVertices[6].z - cp.z) + planes[i].getD() > 0) {
+                + planes[i].getC() * (aabbVertices[6].z - cp.z) + planes[i].getD() > 0) {
                 continue;
             }
             if (planes[i].getA() * (aabbVertices[7].x - cp.x) + planes[i].getB() * (aabbVertices[7].y - cp.y)
-                    + planes[i].getC() * (aabbVertices[7].z - cp.z) + planes[i].getD() > 0) {
+                + planes[i].getC() * (aabbVertices[7].z - cp.z) + planes[i].getD() > 0) {
+                continue;
+            }
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Returns true if this view frustum intersects the given AABB.
+     */
+    public boolean intersects(AABBf aabb) {
```

Comment:
I think as long as `AABB` is being used, we should provide a converter method in `JomlUtil`.
That way we don't have to duplicate the logic like this.